core/src/hlist.rs

```diff
@@ -331,7 +331,7 @@ macro_rules! gen_inherent_methods {
             /// * A single closure (for mapping an HList that is homogenous).
             /// * A single [`Poly`].
             ///
-            /// [`Poly`]: struct.Poly.html
+            /// [`Poly`]: ../traits/struct.Poly.html
```

Comment:
Marginally unrelated question: are these checked by the doc-generator?

Reply:
I installed linkchecker from `rustc/src/tools` and use it religiously

Reply:
To clarify, I think the process is:

```rust
git clone https://github.com/rust-lang/rust
cd rust 
git submodule update # or init, I forget
cd src/tools/linkchecker
cargo install
```

Reply:
@ExpHP Oh; I feel like I just discovered ice-cream. Marvelous!